src/test/java/com/videoclub/suriken/api/MovieControllerTest.java

```diff
@@ -0,0 +1,156 @@
+//        Movie movie3 = new Movie(null, "The Godfather", 1972, Genre.CRIME, "Francis Ford Coppola", 1);
+//        Movie movie4 = new Movie(null, "The Dark Knight", 2008, Genre.ACTION, "Christopher Nolan", 7);
+//    }
+
+    @Test
+    void successfullyAddMovie() {
+        HttpEntity<Movie> request = new HttpEntity<>(new Movie(null, "Titanik", 1995, Genre.DRAMA, "James Camoron", 3));
+        ResponseEntity response = restTemplate.postForEntity("/api/v1/movie", request, Void.class);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+
+        Optional<Movie> respMovie = movieRepository.findMovieByName("Titanik");
+//        logger.info("Id of saved movie is {}", respMovie.get().getId());
+        assertThat(respMovie.isPresent()).isEqualTo(true);
+    }
+
+    @Test
+    void getAMovieWhenExist() {
+        movieRepository.save(new Movie(null, "Titanik", 1995, Genre.DRAMA, "James Camoron", 3));
+
+        ResponseEntity<Movie> response = restTemplate.getForEntity("/api/v1/movie/1", Movie.class);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+    }
+
+    @Test
+    void getAMovieWhenNotExist() {
+        ResponseEntity<Movie> response = restTemplate.getForEntity("/api/v1/movie/1", Movie.class);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.NOT_FOUND);
+    }
+
+    @Test
+    void getAllMovies() {
+        Movie movie1 = new Movie(null, "Titanik", 1995, Genre.DRAMA, "James Camoron", 3);
+        Movie movie2 = new Movie(null, "The Shawshank Redemption", 1994, Genre.DRAMA, "Frank Darabont", 5);
+        Movie movie3 = new Movie(null, "The Godfather", 1972, Genre.CRIME, "Francis Ford Coppola", 1);
+        Movie movie4 = new Movie(null, "The Dark Knight", 2008, Genre.ACTION, "Christopher Nolan", 7);
+
+        List<Movie> movies = Arrays.asList(movie1, movie2, movie3, movie4);
+        movieRepository.saveAll(movies);
+
+        ResponseEntity<Movie[]> response = restTemplate.getForEntity("/api/v1/movie", Movie[].class);
+
+        List<Movie> resMovies = Arrays.asList(response.getBody());
+
+        resMovies.forEach(movie -> logger.debug("Movie name : {}", movie.getName()));
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+
+        AtomicInteger ind = new AtomicInteger();
+        for (Movie mov : movies) {
+            assertThat(resMovies.get(ind.getAndIncrement()).getName()).isEqualTo(mov.getName());
+        }
+    }
+
+    @Test
+    void successfullyRentAMovie() {
+        movieRepository.save(new Movie(null, "Titanik", 1995, Genre.DRAMA, "James Camoron", 3));
+        renterRepository.save(new MovieRenter(null, "Marko", "Lazovic", "marko@gmail.com"));
+
+        HttpEntity<?> req = HttpEntity.EMPTY;
+        ResponseEntity<?> response = restTemplate.exchange("/api/v1/movie/{id}?renterId={renterId}", HttpMethod.PUT, req, Void.class, 1,1);
```

Comment:
Da li znas sta je ? i kako radi?

Reply:
exchange? e pa ne znam ustvari, samo sam procitao da je to preporucen nacin da se izvrsi PUT

Reply:
@dujke011 jel si na exhange mislio?